src/Client.php

```diff
@@ -28,6 +29,14 @@ public function __construct(string $baseUri)
         $this->baseUri = rtrim($baseUri, '/');
     }
 
+    public function setCredentials(string $username, string $password): void
+    {
+        $authHeader = base64_encode("$username:$password");
+        $this->httpClient = (new HttpClientBuilder())
+            ->intercept(new SetRequestHeaderIfUnset('Authorization', 'Basic ' . $authHeader))
+            ->build();
+    }
+
```

Comment:
I don't like it; the Client is being recreated from scratch. What if the constructor had initialized it differently?